Add Metrics support for Azure & GCS Gateway

nitisht opened the pull request:
## Description
We added support for caching and S3 related metrics in #8591. As
a continuation, it would be helpful to add support for Azure & GCS
gateway related metrics as well.

## Motivation and Context
Better Prometheus metrics

## How to test this PR?
Deploy MinIO in Azure Gateway mode and observe Prometheus metrics

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

harshavardhana commented on cmd/bucket-encryption-handlers.go:
```diff
@@ -41,46 +41,46 @@ func (api objectAPIHandlers) PutBucketEncryptionHandler(w http.ResponseWriter, r
 	// Store the bucket encryption configuration in the object layer
 	if err = objAPI.SetBucketSSEConfig(ctx, bucket, encConfig); err != nil {
-		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
+		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, GuessIsBrowserReq(r))
```
You don't need to modify the original function instead create a function alias's in gateway-common.go - this avoids touching elsewhere in the code and keeps the change simple

nitisht replied:
Done, PTAL

Commit:
Add metrics support for Azure & GCS Gateway

We added support for caching and S3 related metrics in #8591. As
a continuation, it would be helpful to add support for Azure & GCS
gateway related metrics as well.

krisis commented on cmd/gateway-common.go:
```diff
@@ -417,3 +417,32 @@ func gatewayHandleEnvVars() {
 	}
 }
+
+// GuessIsS3Req indicates whether incomeing request is S3 API request
```
typo
```suggestion
// GuessIsS3Req indicates whether incoming request is S3 API request
```

vadmeste commented on cmd/gateway-common.go:
```diff
@@ -417,3 +417,32 @@ func gatewayHandleEnvVars() {
 }
+
+// GuessIsS3Req indicates whether incoming request is S3 API request
+func GuessIsS3Req(req *http.Request) bool {
```
Probably a better name is shouldMeterRequest(), I think will look more solid because we are doing `!( .. || .. || )` inside.

nitisht replied:
done

vadmeste reviewed (changes requested):
One comment, also could you update https://docs.min.io/docs/how-to-monitor-minio-using-prometheus.html

vadmeste reviewed (approved):
LGTM
